src/auth-service/utils/request.js

```diff
@@ -55,61 +54,90 @@ const request = {
       //   }
       // });
 
-      const responseFromCreateCandidate = await CandidateModel(tenant).register(
-        request
-      );
+      let filterRequestBody = {};
+      filterRequestBody["body"] = {};
+      filterRequestBody["query"] = {};
+      filterRequestBody["body"] = req;
+      filterRequestBody["query"]["category"] = category;
+
+      const responseFromGenerateCandidateFilter =
+        generateFilter.candidates(filterRequestBody);
+
+      if (responseFromGenerateCandidateFilter.success === false) {
+        callback({
+          success: false,
+          message: "internal server error",
+          errors: { message: responseFromGenerateCandidateFilter.error },
+        });
+      }
 
-      if (responseFromCreateCandidate.success === true) {
-        let createdCandidate = await responseFromCreateCandidate.data;
-        let responseFromSendEmail = await mailer.candidate(
-          firstName,
-          lastName,
-          email,
+      let filter = responseFromGenerateCandidateFilter.data;
+      const responseFromListCandidates = await request.list({
+        tenant,
+        filter,
+      });
+
+      if (responseFromListCandidates.success === true) {
+        callback({
+          success: true,
+          message: "candidate already exists or successfully created",
+          status: httpStatus.OK,
+        });
+      } else if (responseFromListCandidates.success === false) {
+        const responseFromCreateCandidate = await CandidateModel(
           tenant
-        );
-        if (responseFromSendEmail.success === true) {
-          const status = responseFromSendEmail.status
-            ? responseFromSendEmail.status
-            : "";
-          callback({
-            success: true,
-            message: "candidate successfully created",
-            data: createdCandidate,
-            status,
-          });
-        }
+        ).register(req);
+
+        if (responseFromCreateCandidate.success === true) {
+          let createdCandidate = await responseFromCreateCandidate.data;
+          let responseFromSendEmail = await mailer.candidate(
+            firstName,
+            lastName,
+            email,
+            tenant
+          );
+          if (responseFromSendEmail.success === true) {
+            const status = responseFromSendEmail.status
+              ? responseFromSendEmail.status
+              : "";
+            callback({
+              success: true,
+              message: "candidate already exists or successfully created",
```

Comment:
a little change in the response message to handle 2 scenarios:

1. successful creation of candidate
2. detection of duplicate